landing_page/app/controllers/microposts_controller.rb

```diff
@@ -38,7 +38,12 @@ def recover
     helper_method :recover
 
     def index 
-        @microposts = Micropost.kept.order(created_at: :desc) 
+        
+        if params[:archived] == "true"
+            @microposts = Micropost.order(created_at: :desc).kept.where("created_at <= ?", Time.now-7.days)
+        else
+            @microposts = Micropost.kept.where("created_at > ?", Time.now-7.days).order(created_at: :desc)
```

Comment:
```suggestion
        comparison = params[:archived] == "true" ? "<=" : ">"
        query = "created_at #{comparison} ?"
        @microposts = Micropost.order(created_at: :desc).kept.where(query, Time.now-7.days)
```

This is a possible refactoring of this code, with an improve readability.

Reply:
I wouldn't be familiar with the ternary operator but I have seen simple examples of the string interpolation used above. I will look into ternary operators though :)